driver-core/src/main/com/mongodb/connection/SocketSettings.java

```diff
@@ -93,25 +95,27 @@ public Builder applySettings(final SocketSettings socketSettings) {
         /**
          * Sets the socket connect timeout.
          *
-         * @param connectTimeout the connect timeout
+         * @param connectTimeout the connect timeout.
+         * The timeout converted to milliseconds must not be greater than {@link Integer#MAX_VALUE}.
```

Comment:
We had this requirement for forever (the `toIntExact` conversion does not, therefore, introduce a new restriction), but did not document it. Now it is documented.